script/tasks/unpauseAllDiamonds.ts

```diff
@@ -0,0 +1,198 @@
+          const safeInfo = await getSafeInfo(safeAddress, network.name)
+
+          // get a valid nonce
+          const nextNonce = await getNextNonce(
+            pendingTransactions,
+            safeAddress,
+            network.name,
+            chain.id,
+            safeInfo.nonce
+          )
+
+          // prepare SAFE transaction
+          const safeTransaction = await safe.createTransaction({
+            transactions: [
+              {
+                to: diamondAddress as Address,
+                value: 0n,
+                data: calldata,
+                operation: OperationType.Call,
+                nonce: nextNonce,
+              },
+            ],
+          })
+
+          // sign transaction with SAFE_SIGNER_PRIVATE_KEY
+          const signedTx = await safe.signTransaction(safeTransaction)
+          const safeTxHash = await safe.getTransactionHash(safeTransaction)
+
+          // Store transaction proposal in MongoDB
+          try {
+            const result = await storeTransactionInMongoDB(
+              pendingTransactions,
+              safeAddress,
+              network.name,
+              chain.id,
+              signedTx,
+              safeTxHash,
+              senderAddress
+            )
+
+            if (!result.acknowledged) {
+              throw new Error(
+                `[${network.name}] MongoDB insert was not acknowledged`
+              )
+            }
+
+            consola.info('Transaction successfully stored in MongoDB')
+          } catch (error) {
+            consola.error('Failed to store transaction in MongoDB:', error)
+            throw error
+          }
+
+          consola.success(`[${network.name}] Transaction proposed`)
+        } catch (error) {
+          consola.error(
+            `[${network.name}] Error proposing unpause transaction:`,
+            error
+          )
+        }
+      })
+    )
+
+    await mongoClient.close()
+    consola.success('All networks processed successfully.')
+
+    process.exit(0)
+  },
+})
+
+async function getBlacklistedFacetAddresses(
+  networkName: string,
+  blacklistFacets: string
+): Promise<Address[]> {
+  // if blacklist is empty we dont need to look for addresses
+  if (!blacklistFacets) {
+    return []
+  }
+
+  // make sure that networkName is a valid supported chain
+  if (!isValidSupportedChain(networkName)) throw Error('Network ')
+
+  // Split the string into an array of facet names and trim whitespace
+  const facetNames = blacklistFacets.split(',').map((name) => name.trim())
+
+  // Retrieve the corresponding addresses for each facet name from the deploy log
+  const facetAddresses: Address[] = []
+  for (const facetName of facetNames) {
+    try {
+      const facetAddress = await getContractAddressForNetwork(
+        facetName,
+        networkName
+      )
+
+      if (!facetAddress)
+        throw new Error(`No ${facetName} address found in deploy log.`)
```

Comment:
I think this error is redundant as the error is already thrown for `!address` in `getContractAddressForNetwork` function

Reply:
correct, well spotted, thank you